src/main/java/uk/gov/hmcts/reform/unspec/model/BusinessProcess.java

```diff
@@ -1,14 +1,46 @@
 package uk.gov.hmcts.reform.unspec.model;
 
+import com.fasterxml.jackson.annotation.JsonIgnore;
 import lombok.Builder;
 import lombok.Data;
 import uk.gov.hmcts.reform.unspec.enums.BusinessProcessStatus;
 
+import static java.util.Optional.ofNullable;
+
 @Data
 @Builder
 public class BusinessProcess {
 
-    private final String processInstanceId;
-    private final BusinessProcessStatus status;
-    private final String activityId;
+    private String processInstanceId;
+    private BusinessProcessStatus status;
+    private String activityId;
+
+    @JsonIgnore
+    public boolean hasSimilarProcessInstanceId(String processInstanceId) {
+        return this.getProcessInstanceId().equals(processInstanceId);
+    }
+
+    @JsonIgnore
+    public BusinessProcessStatus getStatusOrDefault() {
```

Comment:
Not sure of this method, why would `READY` be default status? Can we just have null check before switch where it's used?

Reply:
this is temporary, was added to test as status will be set to READY in Submitted event callback